src/frame/worker/crypto/recrypt/tests/RecryptWasm.test.ts

```diff
@@ -4,12 +4,40 @@ import * as TestUtils from "../../../../../tests/TestUtils";
 import * as CryptoUtils from "../../CryptoUtils";
 import * as nativePBKDF2 from "../../pbkdf2/native";
 import * as Recrypt from "../RecryptWasm";
+import * as MochRecrypt from "@ironcorelabs/recrypt-wasm-binding";
 
 describe("RecryptWasm", () => {
     beforeAll(() => {
         Recrypt.instantiateApi();
     });
 
+    describe("rotateUsersPrivateKeyWithRetry", () => {
+        const userPrivateKey = new Uint8Array([12, 13, 14]);
+        it("should result in an error when mocked generateKeyPair returns Uint8Array of zeros for augmentationFactor", () => {
+            jest.spyOn(Recrypt.getApi(), "generateKeyPair");
+            Recrypt.rotateUsersPrivateKeyWithRetry(userPrivateKey).engage(
+                (error) => {
+                    expect(error.message).toEqual("Key rotation failed.");
+                },
+                () => fail("Should not invoke success when operation fails")
+            );
+            expect(Recrypt.getApi().generateKeyPair).toHaveBeenCalledTimes(2);
```

Comment:
This assertion should be moved into the reject handler up above on line 20.

Reply:
done